framework/src/utils/PetscSupport.C

```diff
@@ -790,6 +795,18 @@ storePetscOptions(FEProblemBase & fe_problem, const InputParameters & params)
     }
   }
 
+#if !PETSC_VERSION_LESS_THAN(3, 14, 0)
+  for (const auto & reason_flag : reason_flags)
+    // Was the option already found in PetscOptions::flags? Or does it exist in PetscOptions::inames
```

Comment:
You might not need this sentence anymore?

`Was the option already found in PetscOptions::flags?`

Reply:
I think I do. Because a user may put `-snes_converged_reason` or `-ksp_converged_reason` in `petsc_options` in which case I do not want to add `-snes_converged_reason ::failed` or `-ksp_converged_reason ::failed` because the user has indicated they want the convergence reason all the time (and in general we do not want duplicate options)

Reply:
Or am I misunderstanding your question?

Reply:
What I meant was that: I did not see you touch `po.flags` within this loop. That indicates you did not touch `PetscOptions::flags?`. 

From my understanding, you already handled `po.flags` at line 669.

If you did not touch `po.flags`, then you do not need the comment 

`Was the option already found in PetscOptions::flags?`

Reply:
Oh I see your question. I think the comment is still relevant though: `!reason_flag.first` on line 802 is telling us whether the option was already found in `PetscOptions::flags`

Reply:
Oh, I understood now. Thanks